torch/nn/modules/transformer.py

```diff
@@ -159,11 +165,12 @@ class TransformerEncoder(Module):
     """
     __constants__ = ['norm']
 
-    def __init__(self, encoder_layer, num_layers, norm=None):
+    def __init__(self, encoder_layer, num_layers, norm=None, batch_first=False):
```

Comment:
Users may pass a custom encoder/decoder layers here with batch first or not. We need to be careful in that case.

Reply:
What's the best way to handle this? If we add it as an optional argument to the custom encoder/decoder, it may break BC depending on the user's custom encoder/decoder implementation.

Reply:
One way would be to use the `inspect` module's `inspect.getfullargspec` to check if the custom encoder/decoder has the `batch_first` argument, and only pass that argument along to the custom encoder/decoder if it does. This feels inelegant to me, though.